code/game/machinery/vending.dm

```diff
@@ -417,6 +421,35 @@ var/global/num_vending_terminals = 1
 
 //H.wear_id
 
+/**
+ *  Receive payment with cashmoney.
+ *
+ *  usr is the mob who gets the change.
+ */
+/obj/machinery/vending/proc/pay_with_cash(var/obj/item/weapon/spacecash/cashmoney, mob/user)
+	if(!currently_vending) return
+	if(currently_vending.price > cashmoney.get_total())
+		// This is not a status display message, since it's something the character
+		// themselves is meant to see BEFORE putting the money in
+		to_chat(user, "[bicon(cashmoney)] <span class='warning'>That is not enough money.</span>")
+		return 0
+
+	// Bills (banknotes) cannot really have worth different than face value,
+	// so we have to eat the bill and spit out change in a bundle
+	// This is really dirty, but there's no superclass for all bills, so we
+	// just assume that all spacecash that's not something else is a bill
+
+	visible_message("<span class='info'>[usr] inserts a credit chip into [src].</span>")
+	var/left = cashmoney.get_total() - currently_vending.price
+	qdel(cashmoney)
+
+	if(left)
+		dispense_cash(left, src.loc, user)
```

Comment:
I guess this doesn't actually hurt anything, but we don't have a third arg to `dispense_cash()`
What's it do over on Paradise, out of curiosity?